These patches are attributed to filnet - he developed them as part of his work on osgearth.
Details here: https://github.com/filnet/MINGW-packages/blob/74f22c0137d358a8631b915183385251711dd324/mingw-w64-osgearth/0002-geos-3_6_1-support.patch

diff -Naur osgearth-osgearth-2.8-original/src/osgEarthSymbology/GEOS osgearth-osgearth-2.8/src/osgEarthSymbology/GEOS
--- osgearth-osgearth-2.8-original/src/osgEarthSymbology/GEOS	2016-09-15 16:19:21.000000000 +0200
+++ osgearth-osgearth-2.8/src/osgEarthSymbology/GEOS	2017-10-03 21:19:51.166008933 +0200
@@ -26,6 +26,7 @@
 #include <osgEarthSymbology/Style>
 #include <osgEarthSymbology/Geometry>
 #include <geos/geom/Geometry.h>
+#include <geos/geom/GeometryFactory.h>
 
 namespace osgEarth { namespace Symbology
 {
@@ -45,7 +46,7 @@
         void disposeGeometry(geos::geom::Geometry* input);
 
     protected:
-        geos::geom::GeometryFactory* _factory;
+        geos::geom::GeometryFactory::unique_ptr _factory;
     };
 
 } } // namespace osgEarth::Features
diff -Naur osgearth-osgearth-2.8-original/src/osgEarthSymbology/GEOS.cpp osgearth-osgearth-2.8/src/osgEarthSymbology/GEOS.cpp
--- osgearth-osgearth-2.8-original/src/osgEarthSymbology/GEOS.cpp	2016-09-15 16:19:21.000000000 +0200
+++ osgearth-osgearth-2.8/src/osgEarthSymbology/GEOS.cpp	2017-10-03 21:19:51.167008933 +0200
@@ -216,7 +216,7 @@
     geos::geom::PrecisionModel* pm = new geos::geom::PrecisionModel(geom::PrecisionModel::FLOATING);
 
     // Factory will clone the PM
-    _factory = new geos::geom::GeometryFactory( pm );
+    _factory = geos::geom::GeometryFactory::create( pm );
 
     // Delete the template.
     delete pm;
@@ -224,7 +224,6 @@
 
 GEOSContext::~GEOSContext()
 {
-    delete _factory;
 }
 
 geom::Geometry*
@@ -233,12 +232,7 @@
     geom::Geometry* output = 0L;
     if ( input && input->isValid() )
     {
-        output = import( input, _factory );
-
-        // if output is ok, it will have a pointer to f. this is probably a leak.
-        // TODO: Check whether this is a leak!! -gw
-        //if ( !output )
-        //    delete f;
+        output = import( input, _factory.get() );
     }
     return output;
 }
@@ -331,10 +325,7 @@
 {
     if (input)
     {
-        geom::GeometryFactory* f = const_cast<geom::GeometryFactory*>(input->getFactory());
         _factory->destroyGeometry(input);
-        if ( f != _factory )
-            delete f;
     }
 }
 
